mobile_verifier/src/reward_shares.rs

```diff
@@ -107,6 +112,34 @@ impl TransferRewards {
             reward_sum: reward_sum * reward_scale,
         }
     }
+
+    pub fn into_rewards(
+        self,
+        epoch: &'_ Range<DateTime<Utc>>,
+    ) -> impl Iterator<Item = proto::MobileRewardShare> + '_ {
+        let Self {
+            reward_scale,
+            rewards,
+            ..
+        } = self;
+        let start_period = epoch.start.encode_timestamp();
+        let end_period = epoch.end.encode_timestamp();
+        rewards
+            .into_iter()
+            .map(move |(hotspot_key, reward)| proto::MobileRewardShare {
+                start_period,
+                end_period,
+                reward: Some(proto::mobile_reward_share::Reward::GatewayReward(
+                    proto::GatewayReward {
+                        hotspot_key: hotspot_key.into(),
+                        dc_transfer_reward: (reward * reward_scale)
```

Comment:
nit: i think `dc_transfer_amount` works better than `dc_transfer_reward`, ie this is the `amount` for the `reward`, rather than this is the `reward` for the `reward`

Same for `poc_reward` vs `poc_amount`

Its one of those cases of change it now or leave it as once its out there in the wild it gonna be much more difficult to change.